security-analysis/security-analysis-api/src/main/java/com/powsybl/security/distributed/DistributedSecurityAnalysisExecution.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.security.distributed;
+
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.computation.ExecutionEnvironment;
+import com.powsybl.computation.ExecutionHandler;
+import com.powsybl.security.SecurityAnalysisResult;
+import com.powsybl.security.SecurityAnalysisResultWithLog;
+import com.powsybl.security.execution.SecurityAnalysisExecution;
+import com.powsybl.security.execution.SecurityAnalysisExecutionInput;
+
+import java.util.Collections;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Execute a security analysis by spawning a specified number of subtasks, each of which
+ * will consist to a separate call to {@literal itools security-analysis} through the specified
+ * {@link ComputationManager}.
+ *
+ * @author Sylvain Leclerc <sylvain.leclerc at rte-france.com>
+ */
+public class DistributedSecurityAnalysisExecution implements SecurityAnalysisExecution {
+
+    private final ExternalSecurityAnalysisConfig config;
+    private final int subtaskCount;
+
+    public DistributedSecurityAnalysisExecution(ExternalSecurityAnalysisConfig config, int subtaskCount) {
+        this.config = config;
```

Comment:
You should probably assert that config is not null and subtaskCount is positive?